src/Easify.Extensions.Notifications/NotificationService.cs

```diff
@@ -0,0 +1,121 @@
+// This software is part of the Easify framework
+// Copyright (C) 2019 Intermediate Capital Group
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Easify.Extensions.Notifications.Messaging;
+using Easify.Extensions.Notifications.Messaging.Configuration;
+using Easify.Extensions.Notifications.Messaging.Configuration.Validators;
+using Easify.Extensions.Notifications.Messaging.Exceptions;
+using Easify.Extensions.Notifications.Templating;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Easify.Extensions.Notifications
+{
+    public class NotificationService : INotificationService
+    {
+        private readonly IMessagingService _messagingService;
+        private readonly NotificationOptions _options;
+        private readonly ITemplateRenderer _templateRenderer;
+        private readonly ILogger<NotificationService> _logger;
+
+        // TODO: Think on moving this to a dependent service 
```

Comment:
Is this for future or needs to be done now?

Reply:
Let's leave this for now. It's a good one as we shouldn't be new-ing this up. But it's FluentValidation and so static at this time.